yarn-project/acir-simulator/src/client/private_execution.test.ts

```diff
@@ -295,7 +295,7 @@ describe('Private Execution test suite', () => {
       expect(newNullifiers).toEqual(consumedNotes.map(n => n.nullifier));
 
       expect(result.preimages.newNotes).toHaveLength(2);
-      const [recipientNote, changeNote] = result.preimages.newNotes;
```

Comment:
Why this change?

Reply:
The order of the sideffects has changed. There is probably some non-determinism in the compiler if we change the point at which the functions stop being acir :thinking: